SIGSEGV in onAttestation() method

From time to time, I see this nil pointer dereference issue:

```
[2020-04-14 12:14:59]  INFO sync: Requesting parent block currentSlot=686768 parentRoot=bdb62d670fbd
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x4d31c18]

goroutine 273 [running]:
github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).onAttestation(0xc00000c3c0, 0x569b300, 0xc04245e000, 0xc03db25f20, 0x0, 0x0, 0x0, 0x0, 0x0)
        beacon-chain/blockchain/process_attestation.go:129 +0x538
github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).ReceiveAttestationNoPubsub(0xc00000c3c0, 0x569b300, 0xc04242bfb0, 0xc03db25f20, 0x0, 0x0)
        beacon-chain/blockchain/receive_attestation.go:36 +0xee
github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).processAttestation(0xc00000c3c0, 0xc00eb4f740)
        beacon-chain/blockchain/receive_attestation.go:113 +0x543
created by github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).Start
        beacon-chain/blockchain/service.go:233 +0xac5
```

--- a/beacon-chain/blockchain/process_attestation.go
+++ b/beacon-chain/blockchain/process_attestation.go
@@ -70,6 +70,16 @@ func (s *Service) onAttestation(ctx context.Context, a *ethpb.Attestation) ([]ui
 	ctx, span := trace.StartSpan(ctx, "blockchain.onAttestation")
 	defer span.End()
 
+	if a == nil {
+		return nil, errors.New("nil attestation")
+	}
+	if a.Data == nil {
+		return nil, errors.New("nil attestation.Data field")
+	}
+	if a.Data.Target == nil {
+		return nil, errors.New("nil attestation.Data.Target field")
+	}
+
 	tgt := stateTrie.CopyCheckpoint(a.Data.Target)
 	tgtSlot := helpers.StartSlot(tgt.Epoch)
 
@@ -119,12 +129,6 @@ func (s *Service) onAttestation(ctx context.Context, a *ethpb.Attestation) ([]ui
 
 	if indexedAtt.AttestingIndices == nil {
 		return nil, errors.New("nil attesting indices")
-	}
-	if a.Data == nil {
-		return nil, errors.New("nil att data")
-	}
-	if a.Data.Target == nil {
-		return nil, errors.New("nil att target")
 	}
 
 	// Update forkchoice store with the new attestation for updating weight.

--- a/beacon-chain/blockchain/process_attestation_test.go
+++ b/beacon-chain/blockchain/process_attestation_test.go
@@ -124,13 +124,34 @@ func TestStore_OnAttestation(t *testing.T) {
 			wantErr:       true,
 			wantErrString: "does not match current epoch",
 		},
+		{
+			name:          "process nil field (a.Target) in attestation",
+			a:             nil,
+			s:             &pb.BeaconState{},
+			wantErr:       true,
+			wantErrString: "nil attestation",
+		},
+		{
+			name:          "process nil field (a.Data) in attestation",
+			a:             &ethpb.Attestation{},
+			s:             &pb.BeaconState{},
+			wantErr:       true,
+			wantErrString: "nil attestation.Data field",
+		},
+		{
+			name:          "process nil field (a.Target) in attestation",
+			a:             &ethpb.Attestation{Data: &ethpb.AttestationData{}},
+			s:             &pb.BeaconState{},
+			wantErr:       true,
+			wantErrString: "nil attestation.Data.Target field",
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			_, err := service.onAttestation(ctx, tt.a)
 			if tt.wantErr {
-				if !strings.Contains(err.Error(), tt.wantErrString) {
+				if err == nil || !strings.Contains(err.Error(), tt.wantErrString) {
 					t.Errorf("Store.onAttestation() error = %v, wantErr = %v", err, tt.wantErrString)
 				}
 			} else {

